lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorType.java

```diff
@@ -30,6 +30,9 @@
    * Indicates an Evaluator that runs on the CLR.
    */
   CLR,
+  /** Indicates an Evaluator that runs on a local thread.
+   */
+   LOCAL,
```

Comment:
I am adding the LOCAL Evaluator type because I then use it in AllocatatedEvaluatorImpl#makeEvaluatorConfiguration to properly configure the thread mode. In the method the configuration is picked based on Evaluator type.